src/lib.rs

```diff
@@ -173,7 +145,7 @@ impl<'a> Display for SanitizedKey {
 }
 
 struct Serializer {
-    fields: Vec<String>,
+    fields: Vec<(String, String)>,
```

Comment:
I don't think so: `Serializer::emit` internally sanitizes the key and converts the value from the `Display` impl to string.  I can't take references to the results inside `emit`, because then the allocated strings become temporary values and the borrow checker complains.

Perhaps I could put lifetimes onto all the serializer implementation, but that looks like a lot of work and I'm not even sure that it works, and given that it used to be a string before I don't think it's worth it.

Reply:
I made the key a `Cow<'static, str>` so static field names won't require an allocation at least.